invenio_records_rest/config.py

```diff
@@ -58,16 +56,155 @@ def _(x):
         max_result_window=10000,
     ),
 )
+"""Default REST endpoints loaded.
+
+This option can be overwritten to describe the endpoints of the different
+record types. Each endpoint is in charge of managing all its CRUD operations
+(GET, POST, PUT, DELETE, ...).
+
+The structure of the dictionary is as follows:
+
+.. code-block:: python
+
+    from invenio_records_rest.query import es_search_factory
+
+
+    def search_factory(*args, **kwargs):
+        if not current_user.is_authenticated:
+            abort(401)
+        return es_search_factory(*args, **kwargs)
+
+
+    def permission_check_factory():
+        def check_title(record, *args, **kwargs):
+            def can(self):
+                if record['title'] == 'Hello World':
+                    return True
+            return type('Check', (), {'can': can})()
+
+
+    RECORDS_REST_ENDPOINTS = {
+        "record-pid-type": {
```

Comment:
Cosmetics: if we use single quotes in the code consistently, can we do the same in examples?